sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachineScaleSet.java

```diff
@@ -471,8 +471,31 @@ RunCommandResult runShellScriptInVMInstance(
         interface WithGroup extends GroupableResource.DefinitionStages.WithGroup<WithSku> {
         }
 
+        /**
+         * The stage of a virtual machine scale set definition allowing to specify orchestration mode for the virtual machine scale set.
+         *
+         */
+        interface WithOrchestrationMode {
+            /**
+             * Specifies the virtual machine scale set's orchestration mode to be Flexible and fault domain count to default 1.
+             * Virtual machine scale sets with Flexible orchestration allows you to combine the scalability of virtual
+             * machine scale sets in Uniform orchestration mode with the regional availability guarantees of availability sets.
+             * @return The next stage of the definition
+             */
+            DefinitionShared withFlexibleOrchestrationMode();
+
+            /**
+             * Specifies the virtual machine scale set's orchestration mode to be Flexible.
+             * Virtual machine scale sets with Flexible orchestration allows you to combine the scalability of virtual
+             * machine scale sets in Uniform orchestration mode with the regional availability guarantees of availability sets.
+             * @param faultDomainCount By default, when you add a VM to a Flexible scale set, Azure evenly spreads instances across fault domains.
+             * @return The next stage of the definition
+             */
+            DefinitionShared withFlexibleOrchestrationMode(int faultDomainCount);
+        }
+
         /** The stage of a virtual machine scale set definition allowing to specify SKU for the virtual machines. */
-        interface WithSku {
+        interface WithSku extends WithOrchestrationMode {
```

Comment:
why do we need `WithSku extends WithOrchestrationMode`?

Reply:
Client can define a flexible vmss without sku, which has no vm profile in it. In order for them to do it that way, client can skip sku definition when they specify `withFlexibleOrchestrationMode()`, like this:
```java
 VirtualMachineScaleSet vmss = this.computeManager
     .virtualMachineScaleSets()
     .define(vmssName)
     .withRegion(euapRegion)
     .withExistingResourceGroup(resourceGroup)
     .withFlexibleOrchestrationMode()
     .create();
```